### ISSUE TITLE
processing.py and the method of how to get the visualization 

### ISSUE THREAD

**Issue by Tobyzai** [llreda/Stereo_Matching#1]
Hi llreda, thanks for sharing such a helpful project. 

I have two questions:
 1. when using processing.py to get training data and test data, it takes up a lot of space (for example, the unpacked dataset_2 takes up 365G of storage space), is this actually the case? Is there a solution for this? 
 2. What is the method used to generate the visualization in the final GIF motion picture? 
 
I hope to receive your feedback.

**Comment by llreda** [comment]
It's a pleasure to receive your question，
1. The decompressed data set is really large, up to about 1TB， I don't do any additional processing here. Because the tiff file is too large, you can try to store it in 16 bit png or other file formats.
2. I use PCL library to visualize and record the screen, and then convert it into GIF.




---- Replied Message ----
| From | ***@***.***> |
| Date | 10/16/2022 18:02 |
| To | ***@***.***> |
| Cc | ***@***.***> |
| Subject | [llreda/Stereo_Matching] processing.py and the method of how to get the visualization (Issue #1) |

Hi llreda, thanks for sharing such a helpful project.

I have two questions:

when using processing.py to get training data and test data, it takes up a lot of space (for example, the unpacked dataset_2 takes up 365G of storage space), is this actually the case? Is there a solution for this?
What is the method used to generate the visualization in the final GIF motion picture?

I hope to receive your feedback.

**Comment by Tobyzai** [comment]
Thank you so much for such a quick reply. 
Could you elaborate a bit on how to visualize the SCARED2019 dataset using the PCL library? 
Is it visualized using color images (.png) and the corresponding point cloud images (.tiff)? 
Or where can I get more information about it? I'm really sorry, I'm new to this area and still need to keep learning. 
Thank you for your help.

**Comment by llreda** [comment]
Yes, you only need to use color  images and point cloud information to visualize. Using PCL's PointXYZRGB class, you can refer to   https://pcl.readthedocs.io/projects/tutorials/en/latest/pcl_visualizer.html?highlight=pcl_visualizer_demo#compiling -And running the program    
 I'm sorry that I can't find the code at that time.










At 2022-10-16 20:37:06, "Tobyzai" ***@***.***> wrote:

Thank you so much for such a quick reply.
Could you elaborate a bit on how to visualize the SCARED2019 dataset using the PCL library?
Is it visualized using color images (.png) and the corresponding point cloud images (.tiff)?
Or where can I get more information about it? I'm really sorry, I'm new to this area and still need to keep learning.
Thank you for your help.

**Comment by Tobyzai** [comment]
> 

Thank you very much. I will try it. 
Since I am new to the field of 3D reconstruction related, I would appreciate if you could give me some advice on how to learn? What should I learn?

**Comment by Iheckzza** [comment]
Hi !Thank you very much for sharing this amazing work!
I'm studying the matches between origin rbg image and depthmap, so I tried to overlay the rectified rgb image with rectified depthmap, but I found there were some offsets between them, seems that these two are not aligned correctly, I also check the origin depth-rgb pairs and undistorted pairs, the whole dataset 1, keyframe_1 in dataset 2 , whole dataset3 and  datatset 6 looks good, the other pairs exist alignment offsets, each dataset has different kinds of offset In horizontal or vertical direction. I did not calculate the alignment error  accurately but just putting them together and the error is obvious. 

So when I want to get point-matches between two pairs of rgb image(left camera images only) using corresponding point cloud transformation, I always get the wrong points matches. 

The original paper did point out some temporal offset but didn't pull out the solution,so is there any way to align the depth and rgb images? I just want to get each two rgb image(left only) pairs' points-matches with depth transformation. Thank you!
